scripts/keyframes.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/env python3
+"""Given a gfxreconstruct command file, identify "interesting"
+keyframes based on distinct maximal function sets used in a frame.  Keyframes
+selected in this way tend to exercise more distinct parts of the trace,
+as many frames will use the same "footprint" of functions as other frames;
+by collapsing to maximal function sets, keyframes can be selected that more
+thoroughly exercise the graphics API.
+
+The selected keyframes can be further refined through the use of filter
+operations.
+
+To use on a trace captured by gfxreconstruct, the trace must be converted
+to an ASCII command file containing one function or method name per line, e.g.:
+
+    ...
+    vkDestroyImageView
+    vkBeginCommandBuffer
+    vkCmdPipelineBarrier
+    vkCmdWriteTimestamp
+    vkAcquireNextImageKHR
+    vkCmdPipelineBarrier
+    vkCmdBeginRenderPass
+    vkCmdBindPipeline
+    vkCmdBindDescriptorSets
+    vkCmdSetViewport
+    vkCmdBindIndexBuffer
+    vkCmdDrawIndexed
+    vkCmdPipelineBarrier
+    vkCmdWriteTimestamp
+    vkCmdEndRenderPass
+    vkCmdResetQueryPool
+    vkEndCommandBuffer
+    vkFlushMappedMemoryRanges
+    vkResetFences
+    vkQueueSubmit
+    vkQueuePresentKHR
+    ...
+
+The command file must include frame marker functions (e.g. vkQueuePresentKHR
+for Vulkan); if it does not, no frame boundaries can be established, and keyframes
+cannot be calculated.
+
+The gfxrecon-convert tool can be used to create such a file from a trace
+file, using an editor or a script to extract only the function or method
+name associated with each captured call.
```

Comment:
It might be useful to give the sed one-liner that does that.
```
gfxrecon-convert --output stdout vkcube.f1.gfxr | sed "s/.*\"name\":\"vk\([^\"]*\).*/vk\1/" > commandfile.txt
python3 keyframes.py commandfile.txt
```
Assuming this script can read from stdin:
```
gfxrecon-convert --output stdout vkcube.f1.gfxr | sed "s/.*\"name\":\"vk\([^\"]*\).*/vk\1/" | python3 keyframes.py
```

Reply:
My ultimate goal here is to automatically create and parse the output from `gfxrecon-convert`; I'm saving that for a second PR after/if this one is accepted.  But agreed that this revision could have a helpful command line.